app/controllers/coronavirus_pages_controller.rb

```diff
@@ -35,8 +35,36 @@ def publish
     end
   end
 
+  def reorder
+    coronavirus_page
+    @old_positions = positions
+    if request.post? && params.key?(:section_order_save)
+      @new_positions = JSON.parse(params[:section_order_save])
+      set_positions(@new_positions)
+      if draft_updater.send
+        message = { notice: "Sections were successfully reordered." }
+      else
+        set_positions(@old_positions)
+        message = { alert: "Sorry! Sections have not been reordered: " + draft_updater.errors.to_sentence }
+      end
+      redirect_to coronavirus_page_path(slug), message
+    end
+  end
+
 private
 
+  def positions
+    coronavirus_page.sub_sections.each_with_object([]) do |sub_section, array|
+      array << { "id" => sub_section.id, "position" => sub_section.position }
+    end
```

Comment:
I think `map` is a better method to use here. Also, if the positions are memoized, the instance variable `@old_positions` becomes redundant. So
```ruby
def positions
  @positions ||= coronavirus_page.sub_sections.map do |sub_section|
    { "id" => sub_section.id, "position" => sub_section.position }
  end
end
```
The method could be named `old_positions` or `current_positions` to prevent confusion with `positions` being used elsewhere for other collections.